filebeat/input/log/harvester.go

```diff
@@ -100,9 +101,23 @@ type Harvester struct {
 	outletFactory OutletFactory
 	publishState  func(file.State) bool
 
+	metrics   *harvesterProgressMetrics
+	checkSize chan struct{}
+
 	onTerminate func()
 }
 
+// stores the metrics of the harvester
+type harvesterProgressMetrics struct {
+	metricsRegistry             *monitoring.Registry
+	filename                    *monitoring.String
+	started                     *monitoring.String
+	lastPublished               *monitoring.String
+	lastPublishedEventTimestamp *monitoring.String
```

Comment:
instead of serializing timestamps to strings over and over again, it would be nicer if we would introduce a timestamp metric. Then we only need to serialize if there is need (serializing timestamps takes some time + requires allocations).